packages/cardpay-sdk/sdk/currency-utils.ts

```diff
@@ -5,9 +5,39 @@
  */
 
 import BigNumber from 'bignumber.js';
-import { get, isNil } from 'lodash';
+import { get, has, isNil } from 'lodash';
 import supportedNativeCurrencies from './native-currencies';
 
+export const isSupportedCurrency = (nativeCurrency: string) => {
+  return has(supportedNativeCurrencies, `${nativeCurrency}`);
+};
+
+export const isValidAmountForCurrency = (nativeCurrency = 'USD', amount: BigNumberish) => {
+  let currency = get(supportedNativeCurrencies, nativeCurrency);
+  if (!currency) {
+    return false;
+  }
+
+  if (typeof (amount as unknown) === 'string') {
+    amount = Number(amount);
+  } else if (amount instanceof BigNumber) {
+    amount = amount.toNumber();
+  }
+
+  if (typeof amount !== 'number' || isNaN(amount)) {
+    return false;
+  }
+
+  let maxDecimals = currency.decimals;
+  let numberOfDecimals = `${amount}`.split('.')[1]?.length;
+
+  if (numberOfDecimals > maxDecimals) {
+    return false;
+  }
```

Comment:
The immediate context of use for this is a merchant requesting payment from someone. If the amount has too many decimal places for the given currency, we could:
- Not display the amount/not handle the amount, OR
- Round/truncate the amount

@lukemelia do you have any thoughts about what we should do?

Reply:
I'd say whichever is easier from an engineering POV, with a preference for the latter. There is an old adage in software: "Be strict with your output and liberal with your input."

Reply:
Knowing that CS-1736 exists, I think it is better UX to be able to edit a rounded amount!

Reply:
Will remove this method, since the main value of this method was 31-36